components/gitpod-cli/cmd/root.go

```diff
@@ -32,8 +102,101 @@ func Execute() {
 		}
 	}
 
-	if err := rootCmd.Execute(); err != nil {
-		fmt.Println(err)
-		os.Exit(1)
+	err := rootCmd.Execute()
+	exitCode := 0
+	utils.TrackCommandUsageEvent.Outcome = utils.Outcome_Success
+	utils.TrackCommandUsageEvent.Duration = time.Since(time.UnixMilli(utils.TrackCommandUsageEvent.Timestamp)).Milliseconds()
+
+	if err != nil {
+		utils.TrackCommandUsageEvent.Outcome = utils.Outcome_SystemErr
+		exitCode = 1
+		if gpErr, ok := err.(GpError); ok {
+			if gpErr.OutCome != "" {
+				utils.TrackCommandUsageEvent.Outcome = gpErr.OutCome
+			}
+			if gpErr.ErrorCode != "" {
+				utils.TrackCommandUsageEvent.ErrorCode = gpErr.ErrorCode
+			}
+			if gpErr.ExitCode != nil {
+				exitCode = *gpErr.ExitCode
+			}
+		}
+		if utils.TrackCommandUsageEvent.ErrorCode == "" {
+			switch utils.TrackCommandUsageEvent.Outcome {
+			case utils.Outcome_UserErr:
+				utils.TrackCommandUsageEvent.ErrorCode = utils.UserErrorCode
+			case utils.Outcome_SystemErr:
+				utils.TrackCommandUsageEvent.ErrorCode = utils.SystemErrorCode
+			}
+		}
+		if utils.TrackCommandUsageEvent.Outcome == utils.Outcome_SystemErr {
+			errorReport(err)
+		}
+	}
+
+	sendAnalytics()
+
+	if err != nil {
+		fmt.Fprintln(os.Stderr, err)
+		os.Exit(exitCode)
+	}
+}
+
+func sendAnalytics() {
+	if len(utils.TrackCommandUsageEvent.Command) == 0 {
+		return
+	}
+	data, err := utils.TrackCommandUsageEvent.ExportToJson()
+	if err != nil {
+		return
+	}
+	cmd := exec.Command(
+		"/.supervisor/supervisor",
+		"send-analytics",
+		"--event",
+		"gp_command",
+	)
+	cmd.Stdin = strings.NewReader(data)
+	cmd.Stdout = ioutil.Discard
+	cmd.Stderr = ioutil.Discard
+
+	// fire and release
+	_ = cmd.Start()
+	if cmd.Process != nil {
+		_ = cmd.Process.Release()
+	}
+}
+
+func errorReport(err error) {
+	if err == nil {
+		return
+	}
+	reportErrorRequest := &ide_metrics.ReportErrorRequest{
+		ErrorStack: errors.New(err).ErrorStack(),
+		Component:  "gitpod-cli",
+		Version:    gitpod.Version,
+	}
+
+	payload, err := json.Marshal(reportErrorRequest)
+	if err != nil {
+		log.WithError(err).Error("failed to marshal json while attempting to report error")
+		return
+	}
+
+	if err != nil {
+		return
+	}
+	cmd := exec.Command(
+		"/.supervisor/supervisor",
+		"error-report",
+	)
+	cmd.Stdin = bytes.NewReader(payload)
+	cmd.Stdout = ioutil.Discard
+	cmd.Stderr = ioutil.Discard
+
+	// fire and release
+	_ = cmd.Start()
```

Comment:
errors should not be swallowed but logged into os.TempDir()+"/gitpod-cli-errors.log"

the same for send analytics